src/WebJobs.Script/Workers/SharedMemoryDataTransfer/ISharedMemoryManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    public interface ISharedMemoryManager
+    {
+        Task<SharedMemoryMetadata> PutBytesAsync(byte[] content);
+
+        Task<SharedMemoryMetadata> PutObjectAsync(object input);
```

Comment:
For now, if you are adding support only for `byte[]` - remove `PutObjectAsync`

Reply:
The reason I am keeping this is because the caller (`ToRpcInvocationRequest` --> `SharedMemoryDataTransfer` --> `PutObjectAsync`) is dealing with `object` so the checks for which type the `object` is and how to deal with it would be cleaner if kept inside `ISharedMemoryManager`.

Reply:
in that case, remove `PutBytesAsync`